beancount/utils/test_utils.py

```diff
@@ -78,6 +80,18 @@ def subprocess_env():
     return {"PATH": binpath, "PYTHONPATH": find_python_lib()}
 
 
+@contextlib.contextmanager
+def temp_file(suffix: str = ".txt") -> Generator[Path, None, None]:
+    """A context manager that return a filepath inside inside a temporary directory and
+    deletes this directory unconditionally once done.
+
```

Comment:
Can you add under which conditions this is required to use over the vanilla function NamedTemporaryFile() from tempfile?

Reply:
done

Reply:
I think this can be used to fix `test_util.docfile` but it will conflict with other PR.